File: app/src/main/java/com/example/tp4/model/post/retrofit/ApiService.java
package com.example.tp4.model.post.retrofit;

import com.example.tp4.model.post.Post;

import retrofit2.Call;
import retrofit2.http.GET;

import java.util.List;

public interface ApiService {
    @GET("posts")
    Call<List<Post>> getPosts();
}

File: app/src/main/java/com/example/tp4/model/post/retrofit/ApiService.kt
package com.example.tp4.model.post.retrofit

import com.example.tp4.model.post.Post
import retrofit2.Call
import retrofit2.http.GET

interface ApiService {
    @GET("posts")
    fun getPosts(): Call<List<Post>> // Use 'fun' instead of 'Call'
}
